Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +860,71 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getAsString(sectionName: string, settingName: string, resource: any, defaultString: string): string {
```

Comment:
I'm confused -- does this return a string always or a string | undefined?

Reply:
It should always return a string. Not `undefined`

Reply:
What do you mean by "should"? I see it returning `setting.default;`

Reply:
Are there vscode settings that default to `null` or `undefined`? We should be passing in a default value to this function that protects against that too.

Reply:
@bobbrow I don't know about vscode settings -- I was referring to the settings we use that don't currently pass true to this function that can return null.

Reply:
This function is for the VS Code core settings, not the cpptools settings. We pass in a default value to ensure that this function always returns something.

Reply:
Bob is correct, these functions are designed to always return something with hardcoded default values, if section.defaultValue is null, we use the hardcoded value.